### PR TITLE
add reactor feature to micronaut-microservices-services-discover-consul guide

### PR DESCRIPTION
**Pull request by burtbeckwith**

### CODE REVIEW

File: guides/micronaut-microservices-services-discover-consul/bookcatalogue/groovy/src/main/groovy/example/micronaut/Book.groovy [hunk 1]
```diff
@@ -2,12 +2,29 @@ package example.micronaut
...
+
+    @NonNull
+    @NotBlank
+    final String isbn
+
+    @NonNull
+    @NotBlank
+    final String name
+
+    @Creator
```
Comment (sdelamo) [inline guides/micronaut-microservices-services-discover-consul/bookcatalogue/groovy/src/main/groovy/example/micronaut/Book.groovy:24]: Is `@Creator` necessary? If we want to keep it, add a callout.

File: guides/micronaut-microservices-services-discover-consul/bookcatalogue/groovy/src/main/groovy/example/micronaut/BooksController.groovy [hunk 2]
```diff
@@ -8,12 +8,11 @@ import io.micronaut.http.annotation.Get
...
 
-    @Get("/") // <2>
+    @Get // <2>
     List<Book> index() {
-        [
-            new Book("1491950358", "Building Microservices"),
-            new Book("1680502395", "Release It!"),
-            new Book("0321601912", "Continuous Delivery:"),
-        ]
+        Book buildingMicroservices = new Book("1491950358", "Building Microservices")
```
Comment (sdelamo) [inline guides/micronaut-microservices-services-discover-consul/bookcatalogue/groovy/src/main/groovy/example/micronaut/BooksController.groovy]: syntax without local variables is nicer

File: guides/micronaut-microservices-services-discover-consul/bookcatalogue/java/src/main/java/example/micronaut/Book.java [hunk 3]
```diff
@@ -1,24 +1,26 @@
...
     @NonNull
     @NotBlank
+    private final String name;
 
-    private String name;
-
-    public Book() {}
-
-    public Book(@NonNull @NotBlank String isbn, @NonNull @NotBlank String name) {
+    @Creator
```
Comment (sdelamo) [inline guides/micronaut-microservices-services-discover-consul/bookcatalogue/java/src/main/java/example/micronaut/Book.java]: Is `@Creator` necessary? If we want to keep it, add a callout.

File: guides/micronaut-microservices-services-discover-consul/bookcatalogue/kotlin/src/main/kotlin/example/micronaut/Book.kt [hunk 4]
```diff
@@ -1,3 +1,11 @@
 package example.micronaut
 
-data class Book(var isbn: String, val name: String)
\ No newline at end of file
+import io.micronaut.core.annotation.Creator
+import io.micronaut.core.annotation.Introspected
+import javax.validation.constraints.NotBlank
+
+@Introspected
+data class Book @Creator constructor(
```
Comment (sdelamo) [inline guides/micronaut-microservices-services-discover-consul/bookcatalogue/kotlin/src/main/kotlin/example/micronaut/Book.kt]: Is `@Creator` necessary? If we want to keep it, add a callout.

File: guides/micronaut-microservices-services-discover-consul/bookcatalogue/kotlin/src/main/kotlin/example/micronaut/BooksController.kt [hunk 5]
```diff
@@ -7,9 +7,10 @@ import io.micronaut.http.annotation.Get
 class BooksController {
 
     @Get // <2>
-    internal fun index(): List<Book> {
-        return listOf(Book("1491950358", "Building Microservices"),
-                Book("1680502395", "Release It!"),
-                Book("0321601912", "Continuous Delivery:"))
+    fun index(): List<Book> {
+        val buildingMicroservices = Book("1491950358", "Building Microservices")
```
Comment (sdelamo) [inline guides/micronaut-microservices-services-discover-consul/bookcatalogue/kotlin/src/main/kotlin/example/micronaut/BooksController.kt]: syntax without local variables is nicer

File: guides/micronaut-microservices-services-discover-consul/bookinventory/groovy/src/main/groovy/example/micronaut/BookInventory.groovy [hunk 6]
```diff
@@ -2,12 +2,27 @@ package example.micronaut
...
-    String isbn
-    Integer stock
+
+    @NonNull
+    @NotBlank
+    final String isbn
+
+    final int stock
+
+    @Creator
```
Comment (sdelamo) [inline guides/micronaut-microservices-services-discover-consul/bookinventory/groovy/src/main/groovy/example/micronaut/BookInventory.groovy]: Is `@Creator` necessary? If we want to keep it, add a callout.

File: guides/micronaut-microservices-services-discover-consul/bookinventory/java/src/main/java/example/micronaut/BookInventory.java [hunk 7]
```diff
@@ -1,26 +1,24 @@
...
 
-    @NonNull
-    @NotNull
-    private Integer stock;
-
-    public BookInventory() {}
+    private final int stock;
 
-    public BookInventory(@NonNull @NotBlank String isbn, @NonNull @NotNull Integer stock) {
+    @Creator
```
Comment (sdelamo) [inline guides/micronaut-microservices-services-discover-consul/bookinventory/java/src/main/java/example/micronaut/BookInventory.java]: Is `@Creator` necessary? If we want to keep it, add a callout.

File: guides/micronaut-microservices-services-discover-consul/bookinventory/kotlin/src/main/kotlin/example/micronaut/BookInventory.kt [hunk 8]
```diff
@@ -1,3 +1,10 @@
 package example.micronaut
 
-data class BookInventory(var isbn: String, val stock: Int)
\ No newline at end of file
+import io.micronaut.core.annotation.Creator
+import io.micronaut.core.annotation.Introspected
+import javax.validation.constraints.NotBlank
+
+@Introspected
+data class BookInventory @Creator constructor(
```
Comment (sdelamo) [inline guides/micronaut-microservices-services-discover-consul/bookinventory/kotlin/src/main/kotlin/example/micronaut/BookInventory.kt]: Is `@Creator` necessary? If we want to keep it, add a callout.

File: guides/micronaut-microservices-services-discover-consul/bookrecommendation/groovy/src/main/groovy/example/micronaut/Book.groovy [hunk 9]
```diff
@@ -2,12 +2,29 @@ package example.micronaut
...
+
+    @NonNull
+    @NotBlank
+    final String isbn
+
+    @NonNull
+    @NotBlank
+    final String name
+
+    @Creator
```
Comment (sdelamo) [inline guides/micronaut-microservices-services-discover-consul/bookrecommendation/groovy/src/main/groovy/example/micronaut/Book.groovy]: Is `@Creator` necessary? If we want to keep it, add a callout.

File: guides/micronaut-microservices-services-discover-consul/bookrecommendation/groovy/src/main/groovy/example/micronaut/BookRecommendation.groovy [hunk 10]
```diff
@@ -2,11 +2,23 @@ package example.micronaut
...
 @EqualsAndHashCode
+@Introspected
 class BookRecommendation {
-    String name
+
+    @NonNull
+    @NotBlank
+    final String name
+
+    @Creator
```
Comment (sdelamo) [inline guides/micronaut-microservices-services-discover-consul/bookrecommendation/groovy/src/main/groovy/example/micronaut/BookRecommendation.groovy]: Is `@Creator` necessary? If we want to keep it, add a callout.

File: guides/micronaut-microservices-services-discover-consul/bookrecommendation/groovy/src/test/groovy/example/micronaut/BookControllerSpec.groovy [hunk 11]
```diff
@@ -1,31 +1,31 @@
...
+import spock.lang.IgnoreIf
+import spock.lang.Specification
 
 @MicronautTest
 class BookControllerSpec extends Specification {
 
     @Inject
     @Client("/")
-    HttpClient client
+    StreamingHttpClient client
```
Comment (sdelamo) [inline guides/micronaut-microservices-services-discover-consul/bookrecommendation/groovy/src/test/groovy/example/micronaut/BookControllerSpec.groovy:17]: Why did you change from `HttpClient` to `StreamingHttpClient` It seems easier to use `HttpClient::toBlocking` than using StreamingHttpClient and having to wrap it in a Flux

File: guides/micronaut-microservices-services-discover-consul/bookrecommendation/java/src/main/java/example/micronaut/Book.java [hunk 12]
```diff
@@ -11,15 +12,15 @@
...
 
     @NonNull
     @NotBlank
-    private String name;
+    private final String name;
 
-    public Book() {}
-
-    public Book(@NonNull @NotBlank String isbn, @NonNull @NotBlank String name) {
+    @Creator
```
Comment (sdelamo) [inline guides/micronaut-microservices-services-discover-consul/bookrecommendation/java/src/main/java/example/micronaut/Book.java]: Is `@Creator` necessary? If you want to keep, we need to add a callout and explain it to the user.

File: guides/micronaut-microservices-services-discover-consul/bookrecommendation/java/src/main/java/example/micronaut/BookRecommendation.java [hunk 13]
```diff
@@ -10,10 +11,9 @@
 
     @NonNull
     @NotBlank
-    private String name;
-
-    public BookRecommendation() {}
+    private final String name;
 
+    @Creator
```
Comment (sdelamo) [inline guides/micronaut-microservices-services-discover-consul/bookrecommendation/java/src/main/java/example/micronaut/BookRecommendation.java]: Is `@Creator` necessary? If you want to keep, we need to add a callout and explain it to the user.

File: guides/micronaut-microservices-services-discover-consul/bookrecommendation/kotlin/src/main/kotlin/example/micronaut/Book.kt [hunk 14]
```diff
@@ -1,3 +1,11 @@
 package example.micronaut
 
-data class Book(var isbn: String, val name: String)
\ No newline at end of file
+import io.micronaut.core.annotation.Creator
+import io.micronaut.core.annotation.Introspected
+import javax.validation.constraints.NotBlank
+
+@Introspected
+data class Book @Creator constructor(
```
Comment (sdelamo) [inline guides/micronaut-microservices-services-discover-consul/bookrecommendation/kotlin/src/main/kotlin/example/micronaut/Book.kt]: Is `@Creator` necessary? If you want to keep, we need to add a callout and explain it to the user.

File: guides/micronaut-microservices-services-discover-consul/bookrecommendation/kotlin/src/main/kotlin/example/micronaut/BookRecommendation.kt [hunk 15]
```diff
@@ -1,3 +1,8 @@
 package example.micronaut
 
-data class BookRecommendation(val name: String)
\ No newline at end of file
+import io.micronaut.core.annotation.Creator
+import io.micronaut.core.annotation.Introspected
+import javax.validation.constraints.NotBlank
+
+@Introspected
+data class BookRecommendation @Creator constructor(@NotBlank val name: String)
```
Comment (sdelamo) [inline guides/micronaut-microservices-services-discover-consul/bookrecommendation/kotlin/src/main/kotlin/example/micronaut/BookRecommendation.kt]: Is `@Creator` necessary? If you want to keep, we need to add a callout and explain it to the user.

**Review by sdelamo — CHANGES_REQUESTED** [review]
A couple of minor comments.

File: guides/micronaut-microservices-services-discover-consul/bookrecommendation/groovy/src/test/groovy/example/micronaut/BookControllerSpec.groovy (hunk 11 above)
Comment (burtbeckwith) [reply guides/micronaut-microservices-services-discover-consul/bookrecommendation/groovy/src/test/groovy/example/micronaut/BookControllerSpec.groovy:17]: For all of these PRs I followed the guide for Java and made various changes there, then converted the Java to Groovy and Kotlin and merged in the changes to the existing Java and Groovy. In this case the Groovy test was inconsistent so I changed to use StreamingHttpClient like the other two

**Review by sdelamo — APPROVED** [review]

**Pull request merged**